lib/benchee/system.ex

```diff
@@ -32,4 +37,62 @@ defmodule Benchee.System do
     end
   end
 
+  @doc """
+  Returns the number of cores available for the currently running VM.
+  """
+  def num_cores do
+    System.schedulers_online()
+  end
+
+  @doc """
+  Returns an atom representing the platform the VM is running on.
+  """
+  def os do
+    {_, name} = :os.type()
+    os(name)
+  end
+  defp os(:darwin), do: :macOS
+  defp os(:nt), do: :Windows
+  defp os(_), do: :Linux
+
+  @doc """
+  Returns a string with detailed information about the CPU the benchmarks are
+  being performed on.
+  """
+  def cpu_speed, do: cpu_speed(os())
+
+  defp cpu_speed(:Windows), do: "N/A"
+  defp cpu_speed(:macOS), do: system_cmd("sysctl", ["-n", "machdep.cpu.brand_string"])
+  defp cpu_speed(:Linux) do
+    "model name\t:" <> cpu_info = to_string(:os.cmd('cat /proc/cpuinfo | grep "model name"'))
+    String.trim(cpu_info)
+  end
+
+  @doc """
+  Returns an integer with the total number of available memory on the machine
+  running the benchmarks.
+  """
+  def available_memory, do: available_memory(os())
+
+  defp available_memory(:Windows), do: "N/A"
+  defp available_memory(:macOS) do
+    {memory, _} = Integer.parse(system_cmd("sysctl", ["-n", "hw.memsize"]))
+    "#{memory / 1_000_000_000} GB"
+  end
+  defp available_memory(:Linux) do
+    "MemTotal:" <> memory = to_string(:os.cmd('cat /proc/meminfo | grep "MemTotal"'))
```

Comment:
I couldn't for the life of me get the pipe to work with `System.cmd/2`. I asked in Slack, and this is the response I got:

```
System.cmd is much closer to fork/exec that a typical system call in other languages. It does not run a shell so you cannot use shell options. There are a bunch of options ( use Porcelain library, write command to file and call /bin/sh file, :os.cmd in erlang. )
```

So, that's what led me down this path. I could, however, rewrite it to use `System.cmd/2` if you like. The pipe and `grep` were just easier for me at first.